distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso

```diff
@@ -139,44 +153,70 @@ type Statistic
        - statistics: Set of statistics to calculate.
     compute_bulk : Vector -> Vector Statistic -> Vector Any
     compute_bulk data statistics=[Statistic.Count, Statistic.Sum] =
-        count_min_max = statistics.any s->
-            case s of
-                Statistic.Count -> True
-                Statistic.Minimum -> True
-                Statistic.Maximum -> True
-                _ -> False
-
-        java_stats = statistics.map to_moment_statistic
-        skip_java_stats = java_stats.all s->s.is_nothing
-
-        report_invalid _ =
-            statistics.map_with_index i->v->
-                if java_stats.at i . is_nothing then Nothing else
-                    Error.throw (Illegal_Argument.Error ("Can only compute " + v.to_text + " on numerical data sets."))
-        handle_unsupported = Panic.catch Unsupported_Argument_Types.Error handler=report_invalid
-
-        empty_map s = if (s == Statistic.Count) || (s == Statistic.Sum) then 0 else
-            if (s == Statistic.Minimum) || (s == Statistic.Maximum) then Error.throw Empty_Error else
-                Number.nan
-
-        if data.length == 0 then statistics.map empty_map else
-            count_min_max_values = if count_min_max then CountMinMax.new (CountMinMax.toObjectStream data.to_array) Comparator.new else Nothing
-            stats_array = if skip_java_stats then Nothing else
-                handle_unsupported <| Moments.compute data.to_array java_stats.to_array
-
-            statistics.map_with_index i->s->case s of
-                Statistic.Count -> count_min_max_values.count
-                Statistic.Minimum ->
-                    if count_min_max_values.comparatorError then (Error.throw Incomparable_Values) else
-                        count_min_max_values.minimum
-                Statistic.Maximum ->
-                    if count_min_max_values.comparatorError then (Error.throw Incomparable_Values) else
-                        count_min_max_values.maximum
-                Statistic.Covariance s -> calculate_correlation_statistics data s . covariance
-                Statistic.Pearson s -> calculate_correlation_statistics data s . pearsonCorrelation
-                Statistic.Spearman s -> calculate_spearman_rank data s
-                Statistic.R_Squared s -> calculate_correlation_statistics data s . rSquared
-                _ -> stats_array.at i
+        counter = CountMinMax.new
+
+        java_stats = statistics.map .order
```

Comment:
This name seems wrong. It's not a `MomentStatistic` anymore, but just a number representing the order, I'd suggest a rename.